pkg/queue/protobuf_stats_reporter.go

```diff
@@ -67,27 +74,14 @@ func (r *ProtobufStatsReporter) Report(stats network.RequestStatsReport) {
 }
 
 // ServeHTTP serves the stats in protobuf format over HTTP.
-func (r *ProtobufStatsReporter) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	stat := r.stat.Load()
-	if stat == nil {
-		httpError(w, "no metrics available yet")
-		return
-	}
```

Comment:
I still prefer a nil check here as if a `ProtobufStatsReporter` is not created from `NewProtobufStatsReporter`, this could be nil.

Reply:
How can you create it without the new when it has private fields?

Reply:
well I didn't mean actual usage. Just my personal nit that if a function is documented that it can return nil, although we do some steps to ensure it won't return nil actually, I still like to add a gate. :)

Reply:
hm. The actual [example_test for atomic.Value](https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/sync/atomic/example_test.go#L57) uses this exact pattern and does not check for nil. I'd say the idiomatic usage of atomic.Value is to not check nils unnecessarily.

Also, if this was ever nil (which is impossible without programmer error, so a panic is correct) the panic would be  caught by the go http server and recovered with a nice error anyway, so I'm not sure what the value of returning a less descriptive error message instead would be?

Reply:
Good to know that.